third_party/intel/lib/TritonIntelGPUTransforms/OptimizeDotOperands.cpp

```diff
@@ -147,8 +147,8 @@ class FuseTransWithLoad {
     if (!loadOp->hasOneUse() || !loadOpHasBlockIOAttr)
       return false;
 
-    auto ptrType = cast<tt::PointerType>(loadOp.getPtr().getType());
-    if (!isTensorPointerType(ptrType) ||
+    auto ptrType = dyn_cast<tt::PointerType>(loadOp.getPtr().getType());
+    if (!ptrType || !isTensorPointerType(ptrType) ||
         cast<RankedTensorType>(ptrType.getPointeeType()).getRank() != 2)
```

Comment:
which one do you prefer?
```suggestion
    Type ptrType = loadOp.getPtr().getType();
    if (!isTensorPointerType(ptrType) ||
        cast<RankedTensorType>(cast<tt::PointerType>(ptrType).getPointeeType()).getRank() != 2)
```